chain/client/src/view_client.rs

```diff
@@ -505,6 +526,62 @@ impl ViewClientActor {
         cache.push_back(now);
         true
     }
+
+    fn get_chain_info(&mut self) -> ChainInfo {
+        match (|| -> anyhow::Result<ChainInfo> {
+            // Start with the cached self.chain_info. We will update the outdated parts.
+            let mut chain_info = self.chain_info.clone();
+
+            let head = self.chain.head().context("Cannot retrieve chain head")?;
+            chain_info.height = self.get_height(&head);
+
+            // convert config tracked shards
+            // runtime will track all shards if config tracked shards is not empty
+            // https://github.com/near/nearcore/issues/4930
+            chain_info.tracked_shards = if self.config.tracked_shards.is_empty() {
```

Comment:
Probably better to implement a function for this conversion since it's needed in multiple places.

Reply:
obsolete